runtime/tchannel_inbound_call.go

```diff
@@ -134,19 +134,18 @@ func (c *tchannelInboundCall) readReqHeaders(ctx context.Context) error {
 }
 
 // readReqBody reads request body from arg3
-func (c *tchannelInboundCall) readReqBody(ctx context.Context) (wireValue wire.Value, err error) {
+func (c *tchannelInboundCall) readReqBody(ctx context.Context) (stream.Reader, error) {
 	// fail fast if timed out
 	if deadline, ok := ctx.Deadline(); ok && time.Now().After(deadline) {
-		err = context.DeadlineExceeded
-		return
+		return nil, context.DeadlineExceeded
 	}
 
 	treader, err := c.call.Arg3Reader()
 	if err != nil {
 		err = errors.Wrapf(err, "Could not create arg3reader for inbound %s.%s (%s) request",
 			c.endpoint.EndpointID, c.endpoint.HandlerID, c.endpoint.Method,
 		)
-		return
+		return nil, err
```

Comment:
Naked returns were killing the reader cognition in here. Moved to explicit returns.